tests/src/ut_test_kvp.c

```diff
@@ -286,6 +288,26 @@ void test_ut_kvp_bool(void)
     UT_ASSERT( result == false );
 }
 
+void test_ut_kvp_getPlatformProfile( void )
+{
+    ut_kvp_instance_t *pInstance = NULL;
+    bool result;
+
+    pInstance = ut_getPlatformProfile();
+    if(pInstance == NULL)
+    {
+        UT_LOG_ERROR("perhaps platform profile was not passed using -p switch");
+        UT_ASSERT(pInstance == NULL);
```

Comment:
your asserts the wrong way round `UT_ASSERT( pInstance != NULL );`, in this case it will be `NULL` so you want to assert ( I think at least ).

Reply:
After swapping the NULL I noticed you could do this though, since the UT_ASSERT is fall through it just reports errors, you can catch it differently.

```c
 pInstance = ut_getPlatformProfile();
  UT_ASSERT(pInstance != NULL);
  if(pInstance == NULL)
  {
      UT_LOG_ERROR("perhaps platform profile was not passed using -p switch");
      return;
  }
```

Reply:
You could add some steps in, so it's clear for logging, in or add comments to the code.

UT_LOG_STEP("Check ut_getPlatformProfile() - Positive");